src/components/Feed.js

```diff
@@ -1,9 +1,241 @@
+import commentEmpty  from '../comment.png';
+import commentFull  from '../comment_full.png';
+import logo from '../inst_logo.png';
 
-const Feed = () => (
-    <div className="container feed">
-    </div>
-);
+
+async function getListOfPosts(authStr) {
+    return axios.get("http://" + API_DOMAIN + "/api/posts/", { headers: { Authorization: authStr },
+        params: { feed: true,}})
+        .then(result => result.data);
+}
+
+async function getListOfComments(authStr, postId) {
+    return axios.get("http://" + API_DOMAIN + "/api/comments/", { headers: { Authorization: authStr },
+        params: { post_id: postId,}})
+        .then(result => result.data);
+}
+
+function postComment(authStr, credentials) {
+    return axios.post("http://" + API_DOMAIN + "/api/comments/", credentials, {
+        headers: {Authorization: authStr}
+    }).catch((error) => {
+        console.log(error.response);
+    });
+}
+
+function postLike(authStr, postId, isLike) {
+    const postStr = ''.concat(postId);
+    if (isLike) {
+        return axios.post("http://" + API_DOMAIN + "/api/posts/" + postStr + "/like/", {}, {
+            headers: {Authorization: authStr}
+        }).catch((error) => {
+            console.log(error.response);
+        });
+    }
+    else {
+        return axios.post("http://" + API_DOMAIN + "/api/posts/" + postStr + "/dislike/", {}, {
+            headers: {Authorization: authStr}
+        }).catch((error) => {
+            console.log(error.response);
+        });
+    }
+}
+
+function deleteLike(authStr, postId, isLike) {
+    const postStr = ''.concat(postId);
+    if (isLike) {
+        return axios.delete("http://" + API_DOMAIN + "/api/posts/" + postStr + "/like/",  {
+            headers: {Authorization: authStr}
+        }).catch((error) => {
+            console.log(error.response);
+        });
+    }
+    else {
+        return axios.delete("http://" + API_DOMAIN + "/api/posts/" + postStr + "/dislike/", {
+            headers: {Authorization: authStr}
+        }).catch((error) => {
+            console.log(error.response);
+        });
+    }
+}
+
+async function getLikesCount(authStr, postId) {
+    const postStr = ''.concat(postId);
+    return axios.get("http://" + API_DOMAIN + "/api/posts/" + postStr + "/stats/",
+        { headers: { Authorization: authStr }})
+        .then(result => result.data);
+}
+
+function Feed(props) {
+    const { token, setToken } = useToken();
+    const [ postList, setPostList ] = useState([]);
+    const [ loading, setLoading ] = useState(true);
+    const authStr = 'Token '.concat(token);
+
+    useEffect(() => {
+        const handleSome = async () => {
+            setLoading(true);
+            const posts = await getListOfPosts(authStr);
+
+            let index = 0;
+            for (let post of posts) {
+                const likes = await getLikesCount(authStr, post.id);
+
+                posts[index].likesCount = likes.likes_count;
```

Comment:
Doesn't it work this way?
```suggestion
                post.likesCount = likes.likes_count;
```

Reply:
Though it's better to use another approach on the backend and return posts together with their stats

Reply:
Done